### ISSUE TITLE
Provisioning an NFS storage server fails when the `samba` section in `fs.yaml` absent

### ISSUE THREAD

**Issue by pansapiens** [Azure/batch-shipyard#332]
#### Problem Description

Provisioning an NFS storage server fails when calling `shipyard_remotefs_bootstrap.sh` if the `samba` section in `fs.yaml` isn't defined.

#### Batch Shipyard Version

3.9.0

#### Steps to Reproduce

Using this `fs.yml` config:

```yaml
remote_fs:
  resource_group: laxy_resource_group
  location: australiaeast
  managed_disks:
    sku: premium_lrs
    disk_size_gb: 1023
    disk_names:
    - p30-disk0a
    - p30-disk1a
  storage_clusters:
    nfs:
      hostname_prefix: laxy-storage-cluster
      ssh:
        username: shipyard
      file_server:
        type: nfs
        server_options:
          nfs:
            '*':
            - rw
            - no_root_squash
            - no_subtree_check
        mount_options:
        - noatime
        - nodiratime
        mountpoint: /shared
# It appears samba MUST be enabled, otherwise the post-provisioning bootstrap script fails.
#        samba:
#          share_name: shared
      network_security:
        ssh:
        - '*'
      virtual_network:
        name: myvnet
        existing_ok: true
        address_space: 10.0.0.0/8
        subnet:
          address_prefix: 10.0.1.0/24
          name: my-nfs-server-subnet
      public_ip:
        enabled: true
        static: false
      vm_count: 1
      # Available in australiaeast
      vm_size: STANDARD_DS1_v2
      vm_disk_map:
        '0':
          disk_array:
          - p30-disk0a
          - p30-disk1a
          filesystem: btrfs
          raid_level: 0
```

```bash
shipyard fs cluster add nfs -y
```

Fails with:
```
2019-09-02 13:53:39.227 ERROR - Ran out of retry attempts invoking _create_virtual_machine_extension(args=(<azure.mgmt.compute._compute_management_client.ComputeManagementClient object at 0x10bb76320>, RemoteFsSettings(managed_disks=ManagedDisksSettings(location='australiaeast', resource_group='laxy_resource_group', zone=None, sku='premium_lrs', disk_size_gb=1023, disk_provisioned_perf_iops_rw=None, disk_provisioned_perf_mbps_rw=None, disk_names=['p30-disk0a', 'p30-disk1a']), storage_cluster=<convoy.settings.StorageClusterSettings object at 0x10bba7a20>), 'shipyard_remotefs_bootstrap.sh', ['https://laxy.blob.core.windows.net/shipyardremotefs-nfs/shipyard_remotefs_bootstrap.sh', 'https://laxy.blob.core.windows.net/shipyardremotefs-nfs/shipyard_remotefs_addbrick.sh', 'https://laxy.blob.core.windows.net/shipyardremotefs-nfs/shipyard_remotefs_stat.sh'], 'laxy-storage-cluster-vm000', {'p30-disk0a': ('/subscriptions/d200815d-95ed-4701-af52-602be729f1b7/resourceGroups/laxy_resource_group/providers/Microsoft.Compute/disks/p30-disk0a', 'Premium_LRS'), 'p30-disk1a': ('/subscriptions/d200815d-95ed-4701-af52-602be729f1b7/resourceGroups/laxy_resource_group/providers/Microsoft.Compute/disks/p30-disk1a', 'Premium_LRS')}, None, 0, False) kwargs={}) status_code=200
Traceback (most recent call last):
  File "/Users/perry/devel/shipyard/batch-shipyard/.shipyard/lib/python3.7/site-packages/msrestazure/polling/arm_polling.py", line 390, in run
    self._poll()
  File "/Users/perry/devel/shipyard/batch-shipyard/.shipyard/lib/python3.7/site-packages/msrestazure/polling/arm_polling.py", line 418, in _poll
    raise OperationFailed("Operation failed or cancelled")
msrestazure.polling.arm_polling.OperationFailed: Operation failed or cancelled

....

  File "/Users/perry/devel/shipyard/batch-shipyard/shipyard.py", line 1103, in fs_cluster_add
    ctx.blob_client, ctx.config, storage_cluster_id)
  File "/Users/perry/devel/shipyard/batch-shipyard/convoy/fleet.py", line 3018, in action_fs_cluster_add
    storage_cluster_id, _REMOTEFSPREP_FILE[0], _ALL_REMOTEFS_FILES)
  File "/Users/perry/devel/shipyard/batch-shipyard/convoy/remotefs.py", line 843, in create_storage_cluster
    vm_ext = async_ops['vmext'][offset].result()
  File "/Users/perry/devel/shipyard/batch-shipyard/convoy/resource.py", line 104, in result
    return self._op.result()
  File "/Users/perry/devel/shipyard/batch-shipyard/.shipyard/lib/python3.7/site-packages/msrest/polling/poller.py", line 183, in result
    self.wait(timeout)
  File "/Users/perry/devel/shipyard/batch-shipyard/.shipyard/lib/python3.7/site-packages/msrest/polling/poller.py", line 201, in wait
    raise self._exception  # type: ignore
  File "/Users/perry/devel/shipyard/batch-shipyard/.shipyard/lib/python3.7/site-packages/msrest/polling/poller.py", line 152, in _start
    self._polling_method.run()
  File "/Users/perry/devel/shipyard/batch-shipyard/.shipyard/lib/python3.7/site-packages/msrestazure/polling/arm_polling.py", line 400, in run
    raise CloudError(self._response)
msrestazure.azure_exceptions.CloudError: <exception str() failed>
```

NFS cluster node VM extension failure message (eg running shipyard_remotefs_*.sh post-provisioning scripts),
(this can be seen on the NFS node at `sudo cat /var/lib/waagent/custom-script/download/0/stderr` and
`sudo cat /var/log/azure/custom-script/handler.log`):
```
VM has reported a failure when processing extension 'laxy-storage-cluster-vmext000'. Error message: "Enable failed: failed to execute command: command terminated with exit status=3
....
[stderr]
Synchronizing state of nfs-kernel-server.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable nfs-kernel-server
groupadd: invalid group ID ''
".
```

Actual failing command on NFS node is:
```bash
./shipyard_remotefs_bootstrap.sh  -d laxy-storage-cluster -f btrfs -m /shared -o \'*%rw,no_root_squash,no_subtree_check\' -p -r 0 -s nfs -t noatime,nodiratime
```
It seems to be related to not having a 'samba' section in `fs.yaml` (`storage_clusters: nfs: file_server: samba:`).
Once defined, the command on the NFS node becomes (note the additional `-c` option):
```bash
./shipyard_remotefs_bootstrap.sh  -c \'shared:nobody:None:None:None:no:0700:0700\' -d laxy-storage-cluster -f btrfs -m /shared -o \'*%rw,no_root_squash,no_subtree_check\' -p -r 0 -s nfs -t noatime,nodiratime
```

Now it seems to work !

**Comment by alfpark** [comment]
I cannot repro this with `3.9.1`. This specific issue was thought to be fixed in `3.9.0` with commit 35217390889f023cd51213365df605fb36843478.

**Comment by pansapiens** [comment]
Thanks - I tried again with `3.9.1` (without a `samba:` section in `fs.yaml`) and can confirm this is fixed.
